### ISSUE TITLE
not working properly on samsung galaxy s9 android 8

### ISSUE THREAD

**Issue by mtad2002** [staf621/ki4a#29]
when trying to use on samsung galaxy s9 connction close after 2 minutes .

here is app log

[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 korkscrew;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 ssh;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall pdnsd
[libsuperuser][O][SH*] killall: korkscrew: no process killed
[libsuperuser][O][SH*] killall: ssh: no process killed
[libsuperuser][O][SH*] killall: pdnsd: no process killed
[libsuperuser][C][SH%] END
Service received a request! toState=1, current_status=0
Asking for permission to use VPN
Service received a request! toState=4, current_status=2
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/pdnsd -c /data/data/com.staf621.ki4a/ki4a/pdnsd.conf -d; echo $?
[libsuperuser][O][SH-] 0
[libsuperuser][C][SH%] END
Util.startVPN
Starting VPN tunnel
Starting
Protecting SSH fd
Starting new wait4connection
Let's get sshfd
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/sshpass -p ******** /data/data/com.staf621.ki4a/ki4a/bin/ssh 62.113.254.116 -p 22 -l user -NT -g -D 7777 -L 127.0.0.1:8163:4.2.2.4:53 -C -o "ProxyCommand /data/data/com.staf621.ki4a/ki4a/bin/korkscrew --ancillaryfile /data/data/com.staf621.ki4a/ki4a/sshfd_file --directconnection --desthost %h --destport %p" -o "KeepAlive yes" -o "ServerAliveInterval 15" -o "StrictHostKeyChecking=no" -o "GlobalKnownHostsFile=/dev/null"; echo $?
[libsuperuser][O][SH*] Could not create directory '/data/local/home/u0_a471/.ssh'.
Got SSHfd [160]
Configure
Routing all traffic
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/tun2socks --netif-ipaddr 26.26.26.2 --netif-netmask 255.255.255.0 --socks-server-addr 127.0.0.1:7777 --tunmtu 1500 --dnsgw 26.26.26.1:8153 --pid /data/data/com.staf621.ki4a/ki4a/tun2socks.pid; echo $?
[libsuperuser][O][SH-] 0
[libsuperuser][C][SH%] END
Let's send FD to tun2socks
Closing
Closing
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 korkscrew;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 ssh;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 tun2socks/data/data/com.staf621.ki4a/ki4a/bin/busybox killall pdnsd
[libsuperuser][O][SH*] Connection closed by UNKNOWN
[libsuperuser][O][SH-] 255
[libsuperuser][C][SH%] END
Wait4connection: verifying connectivity...
Reconnecting...
[libsuperuser][O][SH*] killall: ssh: no process killed
Util.stopVPN
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 tun2socks
[libsuperuser][O][SH*] killall: tun2socks/data/data/com.staf621.ki4a/ki4a/bin/busybox: no process killed
Stopping VPN tunnel
Closing
ToStatus = 2
[libsuperuser][O][SH*] killall: killall: no process killed
[libsuperuser][C][SH%] END
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall pdnsd; echo $?
[libsuperuser][C][SH%] END
Closing
[libsuperuser][O][SH*] killall: pdnsd: no process killed
[libsuperuser][O][SH-] 1
[libsuperuser][C][SH%] END
Connected
Wait4connection: connection detected [first connection]
ToStatus = 4
Ad loaded callback.
Service received a request! toState=0, current_status=4
Disconnected
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 korkscrew;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 ssh;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 sshpass;/data/data/com.staf621.ki4a/ki4a/bin/busybox killall pdnsd
[libsuperuser][O][SH*] killall: korkscrew: no process killed
[libsuperuser][O][SH*] killall: ssh: no process killed
[libsuperuser][O][SH*] killall: sshpass: no process killed
[libsuperuser][O][SH*] killall: pdnsd: no process killed
[libsuperuser][C][SH%] END
Util.stopVPN
[libsuperuser][C][SH%] START
[libsuperuser][C][SH+] /data/data/com.staf621.ki4a/ki4a/bin/busybox killall -9 tun2socks
[libsuperuser][O][SH*] killall: tun2socks: no process killed
[libsuperuser][C][SH%] END
ToStatus = 0
Stopping VPN tunnel
Closing
Closing
Ad loaded callback.
Ad loaded callback.
Ad loaded callback.
Ad received callback.

**Comment by staf621** [comment]
I'm not completely sure, but looks like the connection is getting closed by some part of the connection, like the server or the channel itself, but does not look like it's being intentionally closed by the app.

Does the SSH connection works well on another client, phone or pc?

**Comment by mtad2002** [comment]
yes it works on many other devices but on this specific device samsung galaxy s9 its not working properly
